### PR TITLE
Ccf upgrade

### PR DESCRIPTION
**Pull request by prakashngit**
This PR introduces changes to upgrade CCF version used for PDO-TP from 0.11.7 to 1.0.19.  A few points to note:

1. CCF Base installation process is different for 1.0.19 (wrt 0.11.7) if one is installing CCF on a bare metal. (without docker) (deb vs tar files). CCF documentation https://microsoft.github.io/CCF/release/1.x/build_apps/install_bin.html should act as your guide. PDO TP Installation expects Ubuntu 20.04.

2. PDO clients no longer need to install CCF Base. Python client package is installed via pip. The client package is tested only on Ubuntu 20.04.

3. CCF user keys are no longer required by PDO clients. PDO clients will instead use one-way TLS. PDO TP now implements a custom authentication policy to permit anonymous PDO clients submit unauthenticated transactions. 

4. PDO-dev docker image uses Ubuntu 20.04 as the base.

5. PDO signing keys while using CCF based PDO TP will now use SECP384R1 as the ECDSA curve.

### CODE REVIEW

**Review by bvavala — COMMENTED** [review]
Overall it looks good.

I can later fix the gipsy and ubuntu 18.04 build/test if we decide so.

**Comment by cmickeyb** [comment]
> Overall it looks good.
> 
> I can later fix the gipsy and ubuntu 18.04 build/test if we decide so.

I would like to see this fixed before we approve. If there is a fundamental problem with Gipsy & 20.04 then we can discuss. Otherwise lets make sure this failure is legitimate and not part of a bigger issue.

**Review by cmickeyb — COMMENTED** [review]
Could you add some detail to the commit messages. Why do we need to skip the host checking? I thought we had already removed that.

**Comment by bvavala** [comment]
Tests in hw mode pass!

File: ccf_transaction_processor/CMakeLists.txt [hunk 1]
```diff
@@ -15,19 +15,18 @@
 cmake_minimum_required(VERSION 3.11)
 
 include(${CCF_DIR}/cmake/preproject.cmake)
-
-project(pdoenc LANGUAGES C CXX)
-
-include(${CCF_DIR}/cmake/ccf.cmake)
+project(logging LANGUAGES C CXX)
```
[the remark this answers is not in the document]
Comment (prakashngit) [reply ccf_transaction_processor/CMakeLists.txt:18]: thanks,  fixed

File: ccf_transaction_processor/CMakeLists.txt [hunk 2]
```diff
@@ -15,19 +15,18 @@
...
 # the PDO TP app will be signed with the RSA key whose locaation is
 # pointed to by PDO_ENCLAVE_CODE_SIGN_PEM
-IF (NOT EXISTS "$ENV{PDO_ENCLAVE_CODE_SIGN_PEM}")
-  MESSAGE(FATAL_ERROR "The enclave signing key pointed to by PDO_ENCLAVE_CODE_SIGN_PEM environment variable does not exist!")
-ENDIF()
-SET(PDO_ENCLAVE_CODE_SIGN_PEM "$ENV{PDO_ENCLAVE_CODE_SIGN_PEM}")
+#IF (NOT EXISTS "$ENV{PDO_ENCLAVE_CODE_SIGN_PEM}")
+#  MESSAGE(FATAL_ERROR "The enclave signing key pointed to by PDO_ENCLAVE_CODE_SIGN_PEM environment variable does not exist!")
+#ENDIF()
+#SET(PDO_ENCLAVE_CODE_SIGN_PEM "$ENV{PDO_ENCLAVE_CODE_SIGN_PEM}")
```
[the remark this answers is not in the document]
Comment (prakashngit) [reply ccf_transaction_processor/CMakeLists.txt:29]: thanks, fixed

File: ccf_transaction_processor/CMakeLists.txt [hunk 3]
```diff
@@ -37,4 +36,4 @@ add_ccf_app(pdoenc
 
 install(TARGETS pdoenc.virtual DESTINATION lib)
 
-sign_app_library(pdoenc.enclave ../etc/oe_sign.conf ${PDO_ENCLAVE_CODE_SIGN_PEM})
+#sign_app_library(pdoenc.enclave ../etc/oe_sign.conf ${PDO_ENCLAVE_CODE_SIGN_PEM})
```
[the remark this answers is not in the document]
Comment (prakashngit) [reply ccf_transaction_processor/CMakeLists.txt:39]: thanks, fixed

File: docker/Dockerfile.pdo-dev [hunk 4]
```diff
@@ -48,8 +48,8 @@
 #   - for develooping based on source in host you might map source into container with an option
 #     like -v $(pwd):/project/pdo/src/private-data-objects/
 
-ARG UBUNTU_VERSION=18.04
-ARG UBUNTU_NAME=bionic
+ARG UBUNTU_VERSION=20.04
```
[the remark this answers is not in the document]
Comment (prakashngit) [reply docker/Dockerfile.pdo-dev:51]: the PR will move pdo end-to-end to 20.04, 18.04 no longer supported.

File: ccf_transaction_processor/README.md [hunk 5]
```diff
@@ -35,64 +36,14 @@ IMPORTANT: When installing CCF and PDO on the same VM for local testing,
...
 
-## Install CCF Base
+## Install CCF
 
-CCF Base with tag 0.11.7 is to be directly installed using the tarball
-from CCF release page.  The following commands will install CCF in the
-folder pointed to be the `CCF_BASE` environment variable.
+Please follow instructions at https://microsoft.github.io/CCF/release/1.x/build_apps/install_bin.html
+to install CCF version 1.0.19. If one follows instructions as is at the above webpage,
+`CCF_BASE` must be set to `/opt/ccf`.
```
[the remark this answers is not in the document]
Comment (prakashngit) [reply ccf_transaction_processor/README.md]: thanks, updated the doc with instructions for installing CCF on bare metal with SIM sgx mode.

**Comment by prakashngit** [comment]
Will create a new PR to replace this one.

**Pull request closed without merging**